dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/v2/DataUploadRunnable.kt

```diff
@@ -54,28 +55,40 @@ internal class DataUploadRunnable(
             val context = contextProvider.context
             // TODO RUMM-0000 it should be already on the worker thread and if readNextBatch is async,
             //  we should wait until it completes before scheduling further
-            val lock = CountDownLatch(1)
-            storage.readNextBatch(
-                noBatchCallback = {
-                    increaseInterval()
-                    lock.countDown()
+            var batchesSent = 0
+            var lastUploadWasSuccess = false
+            do  {
+                val lock = CountDownLatch(1)
+                storage.readNextBatch(
+                    noBatchCallback = {
+                        increaseInterval()
```

Comment:
Question for reviewers: Should we increase the interval only once at the end of the queue if all the batches were sent or should we keep it like this and increase every time a batch was sent (meaning that will increase up to the max limit if all the batches are sent). 
If we go with first option there's less complexity and IMO we are going to keep exactly the same rule as the success rate of the uploads is dictating that. 
If we go with the second option we need to change a bit more this logic and we will also introduce a skew in our upload frequency (if 10 uploads were successful we will only increase the interval once whereas it should be to max frequency)
@0xnm , @xgouchet , @jonathanmos need your opinion here

Reply:
IMO we are changing the definition of the unit-of-work in the uploader, not it will be not a single batch, but a pack of 1-10-100 batches, so we need to change the interval only when unit-of-work is processed.

Reply:
@0xnm I agree but on the other hand if we say handle 100 batches in one job then we will increase the interval only once ? We should have the highest frequency in that case, that's the reason I am willing to keep the first approach in place as it kind of keeps same functionality.

Reply:
I do agree with Nikita, we should keep thinking about unit of work. Extreme case, we upload 100 batches each time, this means we don't need to do an upload cycle as often, since chances are there won't be a 100 batches to upload next.

Reply:
ok...tks all, decision taken, we will go with the unit of work interval decrease/increase. Tks !!